src/nfa/noodlify.cc

```diff
@@ -69,34 +69,34 @@ SegNfa::NoodleSequence SegNfa::noodlify(const SegNfa& aut, const Symbol epsilon,
     // TODO this could probably be written better
     for (auto iter = segments.begin(); iter != segments.end(); ++iter) {
         if (iter == segments.begin()) { // first segment will always have all initial states in noodles
-            for (const State final_state: iter->finalstates) {
+            for (const State final_state: iter->final_states) {
                 SharedPtrAut segment_one_final = std::make_shared<Nfa>(*iter);
-                segment_one_final->finalstates = { final_state };
+                segment_one_final->final_states = {final_state };
                 segment_one_final->trim();
 
-                if (segment_one_final->get_num_of_states() > 0 || include_empty) {
+                if (segment_one_final->size() > 0 || include_empty) {
                     segments_one_initial_final[std::make_pair(unused_state, final_state)] = segment_one_final;
                 }
             }
         } else if (iter + 1 == segments.end()) { // last segment will always have all final states in noodles
-            for (const State init_state: iter->initialstates) {
+            for (const State init_state: iter->initial_states) {
                 SharedPtrAut segment_one_init = std::make_shared<Nfa>(*iter);
-                segment_one_init->initialstates = { init_state };
+                segment_one_init->initial_states = {init_state };
                 segment_one_init->trim();
 
-                if (segment_one_init->get_num_of_states() > 0 || include_empty) {
+                if (segment_one_init->size() > 0 || include_empty) {
                     segments_one_initial_final[std::make_pair(init_state, unused_state)] = segment_one_init;
                 }
             }
         } else { // the segments in-between
-            for (const State init_state: iter->initialstates) {
-                for (const State final_state: iter->finalstates) {
+            for (const State init_state: iter->initial_states) {
+                for (const State final_state: iter->final_states) {
                     SharedPtrAut segment_one_init_final = std::make_shared<Nfa>(*iter);
-                    segment_one_init_final->initialstates = { init_state };
-                    segment_one_init_final->finalstates = { final_state };
+                    segment_one_init_final->initial_states = {init_state };
+                    segment_one_init_final->final_states = {final_state };
                     segment_one_init_final->trim();
 
-                    if (segment_one_init_final->get_num_of_states() > 0 || include_empty) {
+                    if (segment_one_init_final->size() > 0 || include_empty) {
```

Comment:
couldn't size() be cryptic after all?

Reply:
I think it is quite idiomatic in C++ to have size() function. On the other hand, it coud be confusing, since it is not clear what size of an automaton is.

Reply:
`get_num_of_states` is explicit, yet not consistent with the `nfa.increase_size()` operations. `size()` would be cryptic to me if I were to read a random chunk of code and suddenly see the following:
```cpp
...
nfa = foo()
if (nfa.size() < 42) {
    return false;
}
...
```

Why do we return prematurely and unsuccessfully here? What does it mean?

Alternatively,
```cpp
...
nfa = foo()
if (nfa.get_num_of_states() < 42) {
    return false;
}
...
```
reads naturally to me.

As Martin pointed out, size() is idiomatic in C++, but I am not sure how idiomatic it is in the context of automata. Do we usually talk about sizes of automata in the sense of their number of states? Probably, but I have to think about what the name means.

I would keep `size()` for the sake of consistency and idiomatic naming, and further consider adding the current `get_num_of_states()` as an alias for `size()` (be it in its shortened form (`states_num`) or in the original full name).

Reply:
While, it, indeed, is idiomatic, I think it cannot be ambiguous, which in this case is. Since, NFA are basically a composition of several structures, the `size()` can related to several of them (states, transitions). While, most people will think about states, I fear that since there are lots of exots in our community and team, this will not be case.

I would suggest to use `state_size()` and `transition_size()` to avoid this ambiguity.

Reply:
> I would suggest to use state_size() and transition_size() to avoid this ambiguity.

I second this idea. This resolves all the above-mentioned problems and still keeps the idiomatic `size()` in the name.

Reply:
yes yes, good good

Reply:
`states_size` or `state_number`?